product-service/serverless.ts

```diff
@@ -1,72 +1,150 @@
 
 const serverlessConfiguration: Serverless = {
-  service: {
-    name: 'product-service',
-  },
-  frameworkVersion: '2',
-  custom: {
-    webpack: {
-      webpackConfig: './webpack.config.js',
-      includeModules: true
+    service: {
+        name: 'product-service',
     },
+    frameworkVersion: '2',
+    custom: {
+        webpack: {
+            webpackConfig: './webpack.config.js',
+            includeModules: true,
+        },
 
-  },
-  // Add the serverless-webpack plugin
-  plugins: [
-      'serverless-webpack',
-      'serverless-aws-documentation',
-      'serverless-dotenv-plugin'
-  ],
-  provider: {
-    name: 'aws',
-    runtime: 'nodejs12.x',
-    region: 'eu-west-1',
-    apiGateway: {
-      minimumCompressionSize: 1024,
     },
-    environment: {
-      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+    // Add the serverless-webpack plugin
+    plugins: [
+        'serverless-webpack',
+        'serverless-aws-documentation',
+        'serverless-dotenv-plugin',
+    ],
+    provider: {
+        name: 'aws',
+        runtime: 'nodejs12.x',
+        region: config.REGION,
+        apiGateway: {
+            minimumCompressionSize: 1024,
+        },
+        environment: {
+            AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+            SNS_TOPIC_ARN: {
+                Ref: 'SNSTopic'
+            }
+        },
+        iamRoleStatements: [
+            {
+                Effect: 'Allow',
+                Action: 'sns:publish',
+                Resource: {
+                    Ref: 'SNSTopic'
+                }
+            }
+        ],
     },
-  },
-  functions: {
-    getProductList: {
-      handler: 'handler.getProductList',
-      events: [
-        {
-          http: {
-            method: 'get',
-            path: 'products',
-            cors: true
-          }
+    resources: {
+        Resources: {
+            SQSQueue: {
+                Type: 'AWS::SQS::Queue',
+                Properties: {
+                    QueueName: 'catalogItemsQueue',
+                },
+            },
+            SNSTopic: {
+                Type: 'AWS::SNS::Topic',
+                Properties: {
+                    TopicName: 'createProductTopic'
+                }
+            },
+            SNSSubscription: {
+                Type: 'AWS::SNS::Subscription',
+                Properties: {
+                    Endpoint: config.EMAIL,
+                    Protocol: 'email',
+                    TopicArn: {
+                        Ref: 'SNSTopic'
+                    }
+                }
+            },
+            SNSSubscriptionBackup: {
+                Type: 'AWS::SNS::Subscription',
+                Properties: {
+                    Endpoint: config.BACKUP_EMAIL,
+                    Protocol: 'email',
+                    TopicArn: {
+                        Ref: 'SNSTopic'
+                    },
+                    FilterPolicy: {
+                        "tier": ["Mighty"]
```

Comment:
am I missing something? It looks like this filter policy is never invoked as no parameters are passed while publishing SNS notification

Reply:
Hi Oleksandr, thank you for pointing to this issue. It was my mistake, I have uploaded fix

Reply:
Hi @valakar

Thank you for cooperation. Grade is updated.